cineast-core/src/main/java/org/vitrivr/cineast/core/features/OCRSearch.java

```diff
@@ -25,3 +79,221 @@ protected String enrichQueryTerm(String queryTerm) {
     return queryTerm + "~1";
   }
 }
+
+class TesseractRecognizer {
+
+  public String recognizer(BufferedImage image){
+    Tesseract tesseract = new Tesseract();
+    tesseract.setDatapath("resources/OCRSearch/TesseractData/tessdata/");
+    tesseract.setLanguage("eng");
+    tesseract.setTessVariable("tessedit_char_whitelist", "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890");
+    tesseract.setPageSegMode(6);
+    tesseract.setOcrEngineMode(2);
+    tesseract.setTessVariable("user_defined_dpi", "300");
+    try {
+      String result = tesseract.doOCR(image);
+      return result;
+    } catch (TesseractException e) {
+      e.printStackTrace();
+      return null;
+    }
+  }
+}
+
+class AttentionOCRrecognizer{
+  private static final String modelpath = "resources/OCRSearch/AttentionOCRGraph/aocr_frozen_graph/";
+  protected static byte[] graphDef;
+  protected static byte[] imageBytes;
+
+  private static byte[] readAllBytesOrExit(Path path)  {
+    try {
+      return Files.readAllBytes(path);
+    } catch (IOException e) {
+      System.err.println("Failed to read [" + path + "]: " + e.getMessage());
+      System.exit(1);
+    }
+    return null;
+  }
+
+  private static String executeAOCRGraph(byte[] graphDef, Tensor image) {
+    try (Graph g = new Graph()) {
+      g.importGraphDef(graphDef);
+      try (Session s = new Session(g);
+          Tensor result = s.runner().feed("input_image_as_bytes", image).fetch("prediction").run().get(0)) {
+
+        String rstring = new String(result.bytesValue(), Charset.forName("UTF-8"));
+
+        return rstring;
+      }
+    }
+  }
+
+  public String recognizeText(BufferedImage img) throws Exception {
+
+    graphDef = readAllBytesOrExit(Paths.get(modelpath, "frozen_model_temp.pb"));
+
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    ImageIO.write(img, "jpg", baos);
+    imageBytes = baos.toByteArray();
+
+    try (Tensor image = Tensor.create(imageBytes)) {
+      String result = executeAOCRGraph(graphDef, image);
+      return result;
+    }
+    catch (Exception e){
+      e.printStackTrace();
+      return null;
+    }
+  }
+}
+
+class SceneTextOCR{
+  public String TextDetector(BufferedImage image){
+    nu.pattern.OpenCV.loadLocally();
+
+    float scoreThresh = 0.5f;
+    float nmsThresh = 0.1f;
+
+    Mat frame = new Mat(image.getHeight(), image.getWidth(), CvType.CV_8UC3);
+    byte[] data = ((DataBufferByte) image.getRaster().getDataBuffer()).getData();
+    frame.put(0,0, data);
+
+    Net net = Dnn.readNetFromTensorflow("resources/OCRSearch/EASTModel/frozen_east_text_detection.pb");
```

Comment:
Is this already on data.vitrivr.org? If so, it should be added to externalFiles.csv